cpp/ql/src/Architecture/General Top-Level Information/GeneralStatistics.ql

```diff
@@ -5,31 +5,57 @@
  * @id cpp/architecture/general-statistics
  * @tags maintainability
  */
+
 import cpp
 
 from string l, string n
-where (l = "Number of Namespaces" and
-         n = count(Namespace p | p.fromSource()).toString())
-   or (l = "Number of Files" and
-         n = count(File f | f.fromSource()).toString())
-   or (l = "Number of Header Files" and
-         n = count(HeaderFile f | f.fromSource()).toString())
-   or (l = "Number of C Files" and
-         n = count(CFile f | f.fromSource()).toString())
-   or (l = "Number of C++ Files" and
-         n = count(CppFile f | f.fromSource()).toString())
-   or (l = "Number of Classes" and
-         n = count(Class c | c.fromSource() and not c instanceof Struct).toString())
-   or (l = "Number of Structs" and
-         n = count(Struct s | s.fromSource()and not s instanceof Union).toString())
-   or (l = "Number of Unions" and
-         n = count(Union u | u.fromSource()).toString())
-   or (l = "Number of Functions" and
-         n = count(Function f | f.fromSource()).toString())
-   or (l = "Number of Lines Of Code" and
-         n = sum(File f, int toSum | (f.fromSource()) and (toSum = f.getMetrics().getNumberOfLinesOfCode()) | toSum).toString())
-   or (l = "Self-Containedness" and
-         n = (100 * sum(Class c, int toSum | (c.fromSource()) and (toSum = c.getMetrics().getEfferentSourceCoupling()) | toSum)
-                  / sum(Class c, int toSum | (c.fromSource()) and (toSum = c.getMetrics().getEfferentCoupling()) | toSum)).toString()
-           + "%")
+where
+  l = "Number of Namespaces" and
+  n = count(Namespace p | p.fromSource()).toString()
+  or
+  l = "Number of Files" and
+  n = count(File f | f.fromSource()).toString()
+  or
+  l = "Number of Header Files" and
+  n = count(HeaderFile f | f.fromSource()).toString()
+  or
+  l = "Number of C Files" and
+  n = count(CFile f | f.fromSource()).toString()
+  or
+  l = "Number of C++ Files" and
+  n = count(CppFile f | f.fromSource()).toString()
+  or
+  l = "Number of Classes" and
+  n = count(Class c | c.fromSource() and not c instanceof Struct).toString()
+  or
+  l = "Number of Structs" and
+  n = count(Struct s | s.fromSource() and not s instanceof Union).toString()
+  or
+  l = "Number of Unions" and
+  n = count(Union u | u.fromSource()).toString()
+  or
+  l = "Number of Functions" and
+  n = count(Function f | f.fromSource()).toString()
+  or
+  l = "Number of Lines Of Code" and
+  n = sum(File f, int toSum |
+      f.fromSource() and toSum = f.getMetrics().getNumberOfLinesOfCode()
+    |
+      toSum
+    ).toString()
+  or
+  l = "Self-Containedness" and
+  n = (
+        100 *
+            sum(Class c, int toSum |
+              c.fromSource() and toSum = c.getMetrics().getEfferentSourceCoupling()
+            |
+              toSum
+            ) /
+          sum(Class c, int toSum |
+            c.fromSource() and toSum = c.getMetrics().getEfferentCoupling()
+          |
+            toSum
+          )
+      ).toString() + "%"
```

Comment:
It's not a big deal, but I'm not especially keen on what's happened with the arithmetic here.

Reply:
Fixed by rewriting the expressions to fit within the 100-colum limit. That should be correct according to https://semmle.slack.com/archives/C20TR4EAZ/p1555058003000800.